src/cloudinary-image.component.ts

```diff
@@ -1,41 +1,60 @@
 import {
-    Component,
-    ElementRef,
-    EventEmitter,
-    Input,
-    Output,
-    ContentChildren,
-    QueryList,
-    AfterViewInit,
-    OnInit,
-    OnChanges,
-    SimpleChanges,
-    OnDestroy
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ContentChildren,
+  QueryList,
+  AfterViewInit,
+  OnInit,
+  OnChanges,
+  SimpleChanges,
+  OnDestroy, ContentChild,
 } from '@angular/core';
 import { Cloudinary } from './cloudinary.service';
 import { CloudinaryTransformationDirective } from './cloudinary-transformation.directive';
+import { CloudinaryPlaceHolder } from './cloudinary-placeholder.component';
 import { isBrowser } from './cloudinary.service';
 
 @Component({
   selector: 'cl-image',
-  template: '<img>'
+  template: `<img [style.display]="shouldShowPlaceHolder ? 'none' : 'inline'" (load)="hasLoaded()">
+  <div [style.display]="shouldShowPlaceHolder ? 'inline' : 'none'">
+      <ng-content></ng-content>
+  </div>
+  `,
 })
 export class CloudinaryImage
-  implements AfterViewInit, OnInit, OnChanges, OnDestroy {
+  implements AfterViewInit, OnInit, AfterViewInit, OnChanges, OnDestroy {
   @Input('public-id') publicId: string;
   @Input('client-hints') clientHints?: boolean;
+  @Input('loading') loading: string;
+  @Input('width') width?: string;
+  @Input('height') height?: string;
 
   @ContentChildren(CloudinaryTransformationDirective)
   transformations: QueryList<CloudinaryTransformationDirective>;
+  @ContentChild(CloudinaryPlaceHolder) placeholderComponent: CloudinaryPlaceHolder;
 
   @Output() onLoad: EventEmitter<boolean> = new EventEmitter(); // Callback when an image is loaded successfully
   @Output() onError: EventEmitter<boolean> = new EventEmitter(); // Callback when an image is loaded with error
 
   observer: MutationObserver;
+  shouldShowPlaceHolder: boolean = true;
 
   constructor(private el: ElementRef, private cloudinary: Cloudinary) {}
 
   ngOnInit(): void {
+    if (this.width && this.placeholderComponent) {
+      this.placeholderComponent.setWidth(this.width);
+    }
+    if (this.height && this.placeholderComponent) {
+      this.placeholderComponent.setHeight(this.height);
+    }
+    if (this.placeholderComponent) {
+      this.placeholderComponent.setPublicId(this.publicId);
+    }
```

Comment:
@strausr , how do we FW the transformation to the placeholder?

Reply:
In the image component- https://github.com/cloudinary/cloudinary_angular/blob/02d879152a0648927a24dbc32392ae1249ece5e4/src/cloudinary-image.component.ts#L120